katunilya/pymon
# v0.1.0

- Rename `Pipe` to `pipe` (#75)
- Rename `Future` to `future` (#75)
- Remove `then` and `then_async` from `pipe` (#75)
- Remove `then` and `then_async` from `future` (#75)
- Fix `pipe` and `future` in `README.md` (#75)
- Update docs (#75)
- Rename `this_async` to `this_future` (#76)
- Move `returns_future` to `future.returns` (#76)
- Move `pipeline` to `pipe.returns` (#76)
- Rename `returns_async` to `returns_future` (#76)
- Wrap `this_future` and `returns_future` with `future.returns` decorator (#76)
- Improve typing for `future.returns` (#76)
- Fix `future.returns` usage in `pymon.result` (#76)
- Remove `pipeline` and `future_returns` from pymon root import (#76)
- Change signature for `this` and `this_future` (#76)
- Fix `Func` and `FutureFunc` to `compose` and `compose_future` (#76)
- Add typings for `this` and `this_future` (#76)
- Delete `func` and `future_func` (#76)
- Add example for `compose` in docstring (#76)
- Add example for `compose_future` in docstring (#76)
- Rework `compose` and `compose_future` to be more optimal (#76)
- Add tests for `compose` with sync-only functions (#76)
- Add `compose` tests for async-only (#76)
- Add test for `compose` of both sync and async functions (#76)
- Fix wrong `pipe`/`future` usage in `compose_future` (#76)
- Make `compose_future` private in module (#76)
- Remove old `Predicate` and `FuturePredicate` (#81)
- Add `each` and `_each_future` predicate composition classes (#81)
- Add tests for for `each` (#81)
- Add `one` and `_one_future` class for predicate disjunction (#81)
- Add tests for `one` predicate composition class (#81)
- Add tests for empty predicate compositions  (#81)
- Add example in docstring for `each` and `one` (#81)
- Replace static error with error factory function (#83)
- Remove old `check` and `check_future` (#83)
- Rename `ok_when` and `ok_when_future` to `check` and `check_future`  (#83)
- Add examples in docstring for `check` and `check_future` (#83)
- Reimplement `choose_ok` as class (#79)
- Update tests for `choose_ok` (#79)
- Add tests for failed `choose_ok` situations (#79)
- Implement `choose_ok_future` as class (#80)
- Add tests for `choose_ok_future` (#80)
- Add `choose_some` combinator (#55)
- Add tests for `choose_some` (#55)
- Add `choose_some_future` combinator (#78)
- Add tests for `choose_some_future` (#78)
- Add `Entity` protocol class (#84)
- Add `Signed` protocol class (#84)
- Add `Auditable` protocol class (#84)
- Add test for `future` (#30)
- Add test for `future.returns` (#30)
- Add test for `future.__lshift__` (#30)
- Add test for `future.__rshift__` (#30)
- Add tests for `pipe.__lshift__` (#30)
- Add test for `pipe.__rshift__` (#30)
- Add test for `pipe.finish` (#30)
- Add test for `pipe.returns` (#30)
- Add test for `this` (#30)
- Add test for `this_future` (#30)
- Add test for `returns` (#30)
- Add tests for `returns_future` (#30)
- Fix todos for `maybe` tests (#30)
- Add test for `if_some` (#30)
- Add tests for `if_none` (#30)
- Add tests for `if_none_returns` (#30)
- Rename `check` to `ok_when` (#95)
- Reimplemnt `some_when` via if (#96)
- Add docstring example for `some_when` (#96)
- Add `some_when_future` (#96)
- Add test for `some_when_future` (#96)
- Rename `check` to `ok_when` (#95)
- Reimplemnt `some_when` via if (#96)
- Add docstring example for `some_when` (#96)
- Add `some_when_future` (#96)
- Add test for `some_when_future` (#96)
- Add test for `some_when` (#30)
- Fix todos in tests for maybe (#30)
- Add `if_ok_returns` (#30)
- Add tests for `if_ok`, `if_error`, `if_ok_returns`, `if_error_returns` (#30)
- Add test for `ok_when`, `ok_when_future` (#30)
- Fix try-except for `safe` and `safe_future` (#30)
- Add test for `safe` (#30)
- Add test for `safe_future` (#30)
- Add codecov config (#30)
- Add code coverage report generation (#30)
- Add example for `future.returns` (#77)
- Add example docstring for `pipe.returns` (#77)
- Add example docstring for `pipe.finish` (#77)
- Add example docstring for `this` (#77)
- Add example docstring for `this_future` (#77)
- Add docstring for `returns` (#77)
- Add example docstring for `returns_future` (#77)
- Add example docstring for `safe_future` (#77)
- Add example docstring for `safe` (#77)
- Add example docstring for `if_error_returns` (#77)
- Add example docstring for `if_ok_returns` (#77)
- Add example docstring for `if_error` (#77)
- Add example docstring for `if_ok` (#77)
- fixup! Add example docstring for `if_ok` (#77)
- fixup! fixup! Add example docstring for `if_ok` (#77)
- Add example docstring for `if_some` (#77)
- Add example docstring for `if_none` (#77)
- Add example docstring for `if_none_returns` (#77)
- Add example docstring for `if_some_returns` (#77)
- Remove docstring for `_compose_future` utility class (#77)
- Fix `pipe` name in `README.md` line 136 (#77)
- Improve `README.md` (#77)
- Regenerate docs (#77)
- Add licence (#85)
- Add `pytest-cov` (#85)
- Add action for publishing package to PyPI (#85)
- Rename project to `fundom` (#85)
- Bump version v0.0.8->v0.1.0
